tracer/src/Datadog.Trace/AppSec/Transports/Http/HttpTransport.Core.cs

```diff
@@ -40,6 +51,48 @@ public IHeadersCollection GetResponseHeaders()
         {
             return new HeadersCollectionAdapter(_context.Response.Headers);
         }
+
+        public void WriteBlockedResponse(string templateJson, string templateHtml)
+        {
+            _context.Items["block"] = true;
+            var httpResponse = _context.Response;
+            httpResponse.Clear();
+            httpResponse.Headers.Clear();
+            httpResponse.StatusCode = 403;
+            var syncIOFeature = _context.Features.Get<IHttpBodyControlFeature>();
+            if (syncIOFeature != null)
+            {
+                // allow synchronous operatoin for net core >=3.1 otherwise invalidoperation exception
+                syncIOFeature.AllowSynchronousIO = true;
+            }
+
+            var template = templateJson;
+            if (_context.Request.Headers["Accept"] == "application/json")
+            {
+                httpResponse.ContentType = "application/json";
+            }
+            else
+            {
+                httpResponse.ContentType = "text/html";
+                template = templateHtml;
+            }
```

Comment:
Personally, I would probably reverse this condition and use "contains" e.g. if the Accept header contains "text/html", use that, otherwise return json. Otherwise, if they're using a json derivative (e.g. `applicataion/hal+json`) we incorrectly send them text/html.